### PR TITLE
Remove some code duplication in bindings

### PR DESCRIPTION
**Pull request by dyashuni**
Introduced new functions `get_input_array_shapes` and `get_input_ids_and_check_shapes` to remove code duplication
Small refactoring: added spaces

### CODE REVIEW

**Comment by yurymalkov** [comment]
Thanks! Proposed few cosmetic changes

**Review by yurymalkov — COMMENTED** [review]
Oops. Forgot to send the review

File: python_bindings/bindings.cpp [hunk 1]
```diff
@@ -79,6 +79,45 @@ inline void assert_true(bool expr, const std::string & msg) {
...
+    std::vector<size_t> ids;
+    if (!ids_.is_none()) {
+        py::array_t < size_t, py::array::c_style | py::array::forcecast > items(ids_);
+        auto ids_numpy = items.request();
+        // check shapes
+        bool valid = false;
+        if ((ids_numpy.ndim == 1 && ids_numpy.shape[0] == rows) || (ids_numpy.ndim == 0 && rows == 1)) {
+            valid = true;
+        }
+        if (!valid) throw std::runtime_error("wrong dimensionality of the labels");
```
[the remark this answers is not in the document]
Comment (dyashuni) [reply python_bindings/bindings.cpp]: I also added logging of actual input label shape and input data vector shape in the error message.

**Review by yurymalkov — APPROVED** [review]
Looks good to me!

**Pull request merged**